### ISSUE TITLE
Recurring payments

### ISSUE THREAD

**Issue by fallerOfFalls** [BananoCoin/natrium-android-wallet#15]
How hard would it be to implement a automatic recurring payments feature? What are the obstacles?

Does the wallet have unencrypted access to my seed at all times (or is it somehow encrypted by the pin)? Could it just do the payment and notify me about it?

**Comment by bbedward** [comment]
The trick is that, all block creation/signing happens on device - so your private key is never sent to the server. The server is only used for block publishing.

If the server knew your private key, that would probably be the most reliable place to implement it. Since it doesn't, the alternative would probably be to schedule the payments on device - probably using JobScheduler or Alarm manager.

You just lose a little bit of reliability when it's all on device, like if the app is force stopped then the scheduled job may never run.

**Comment by fallerOfFalls** [comment]
Yes I mean the version where all the signing happens on the device.

Could the server be aware of what the schedule is and then send a push notification if the scheduled payment didn't happen? I'd be okay with non-absolute reliability and notification when it fails, but having it normally take care of things by itself would be great.

Or is there some way for the server to start the app and trigger the app to do the payment? The app still checks it's own payment schedule and only signs if it all checks out, but the server would be responsible for 'reminding' the app. I don't know enough about android to know whether this is possible.

Of course, there would still be situations where it wouldn't work, if your phone has no internet and also the app is stopped.
